agents/domains/evm/lightinbox.go

```diff
@@ -61,3 +62,70 @@ func (a lightInboxContract) SubmitAttestation(
 
 	return tx, nil
 }
+
+//nolint:dupl
+func (a lightInboxContract) SubmitStateReportWithSnapshot(transactor *bind.TransactOpts, stateIndex int64, signature signer.Signature, snapPayload []byte, snapSignature []byte) (tx *ethTypes.Transaction, err error) {
+	rawSig, err := types.EncodeSignature(signature)
+	if err != nil {
+		return nil, fmt.Errorf("could not encode signature: %w", err)
+	}
+
+	// TODO: Is there a way to get a return value from a contractTransactor call?
```

Comment:
Not something we have to fix for this pr, but I am exploring a few soltuions here:

https://twitter.com/trajan0x/status/1701008059991232897